src/components/Img.tsx

```
@@ -3,18 +3,24 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
  * Image component that automatically handles baseUrl for Docusaurus.
  * Use this instead of <img> tags in MDX files to ensure images work
  * correctly with arbitrary baseUrl configurations.
+ * 
+ * Accepts an optional `prompt` attribute for AI-generated images,
+ * which stores the generation prompt as a data attribute.
  */
-export default function Img({ src, ...props }: ImgProps) {
+export default function Img({ src, prompt, ...props }: ImgProps) {
   // If src starts with /, it's an absolute path from static folder
   // useBaseUrl will prepend the baseUrl automatically
   const imageSrc = src.startsWith('/') ? useBaseUrl(src) : src;
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)